doc/sphinx/proof-engine/proof-handling.rst

```diff
@@ -539,6 +539,112 @@ Requesting information
 
+.. _showing_diffs:
+
+Showing differences between proof steps
+---------------------------------------
+
+
+Coq can automatically highlight the differences between successive proof steps.
+For example, the following screen shots of coqtop and CoqIDE show the same
+“intros” tactic.  This tactic creates two new hypotheses, highlighted in green.
+The conclusion is entirely in pale green because it’s changed, but no tokens were added
+to it.  The third screen shot uses the "removed" option, so it shows the conclusion a
+second time with the old text, with deletions marked in red.  Also, since the hypotheses are
+new, no line of old text is shown for them.
+
+.. _diffs_coqtop_on:
+
+  .. image:: ../_static/diffs-coqtop-on.png
+     :alt: coqtop with Set Diffs on
+
+.. _diffs_coqide_on:
+
+  .. image:: ../_static/diffs-coqide-on.png
+     :alt: |CoqIDE| with Set Diffs on
+
+.. _diffs_coqide_removed:
+
+  .. image:: ../_static/diffs-coqide-removed.png
+     :alt: |CoqIDE| with Set Diffs removed
+
+**How to enable diffs**
+
+.. opt:: Set Diffs %( "on" %| "off" %| "removed" %)
+.. This ref doesn't work: :opt:`Set Diffs %( "on" %| "off" %| "removed" %)`
+
+The “on” option highlights added tokens in green, while the “removed” option
+additionally reprints items with removed tokens in red.  Unchanged tokens in
+modified items are shown with pale green or red.  (Colors are user-configurable.)
+
+For coqtop, showing diffs can be enabled when starting coqtop with the
+``-diffs on|off|removed`` command-line option or with the ``Set Diffs``
+command within Coq.  You will need to provide the ``-color on|auto`` command-line option when
+you start coqtop in either case.
+
+Colors for coqtop can be configured by setting the ``COQ_COLORS`` environment
+variable, which has the same format as the GNU ``LS_COLORS`` variable.  The string
+uses ANSI escape codes to represent attributes.  For example:
+
+        ``export COQ_COLORS=”diff.added=4;48;2;0;0;240:diff.added.bg=44”``
+
+sets the highlights for added text to underlined (the 4) with a background RGB
+color (0, 0, 240) and unmodified text in a modified item to a blue background.
+The other tags used by diffs are ``diff.removed`` and ``diff.removed.bg``.  Note
+that if you specify ``COQ_COLORS``, the predefined attributes are ignored.
+
+
+In CoqIDE, diffs should be enabled from the View menu.  Don’t use the ``Set Diffs``
+command in CoqIDE.  You can change the background colors shown for diffs from the
+Edit | Preferences | Tags panel by changing the settings for the ``diff.added``,
+``diff.added.bg``, ``diff.removed`` and ``diff.removed.bg`` tags.  This panel also
+allows let you control other attributes of the highlights, such as the foreground
+color, bold, italic, underline and strikeout.
+
+Note: As of this writing (August 2018), Proof General will need minor changes
+to be able to show diffs correctly.  We hope it will support this feature soon.
+
+**How diffs are calculated**
```

Comment:
This should be a title, rather than bold text, I think

Reply:
It should be subordinate to the subsection "Showing differences between proof steps".  I tried making this a subsubsection but Sphinx gave an error for the input shown below.  

```
How diffs are calculated
^^^^^^^^^^^^^^^^^^^^^^^^
```

Reply:
Which error?

Reply:
```
Warning, treated as error:
/cygdrive/c/Users/Jim.DESKTOP-MMPNVGH/Documents/coqSource/coq/doc/sphinx/proof-engine/proof-handling.rst:569:Title level inconsistent:

How to enable diffs
^^^^^^^^^^^^^^^^^^^
make[1]: *** [Makefile.doc:59: sphinx] Error 2
```

Reply:
Use:

    How diffs are calculated
    ````````````````````````

because this is how the previous sub-section work in this chapter.